### ISSUE TITLE
TryToResolveDnnCodeToLink fails if a page has been removed

### ISSUE THREAD

**Issue by pascalmoser** [2sic/2sxc#758]
If 2sxc tries to resolve a Page:[id] token and the page with this id does not exist, an error message appears: 

```
Beim Rendern des Templates ist ein Fehler aufgetreten: System.NullReferenceException: Object reference not set to an instance of an object.
 at DotNetNuke.Entities.Urls.AdvancedFriendlyUrlProvider.FriendlyUrlInternal(TabInfo tab, String path, String pageName, String portalAlias, PortalSettings portalSettings)
 at DotNetNuke.Entities.Urls.AdvancedFriendlyUrlProvider.FriendlyUrl(TabInfo tab, String path, String pageName, PortalSettings portalSettings)
 at DotNetNuke.Common.Globals.NavigateURL(Int32 tabID, Boolean isSuperTab, PortalSettings settings, String controlKey, String language, String pageName, String[] additionalParameters)
 at DotNetNuke.Common.Globals.NavigateURL(Int32 tabID, Boolean isSuperTab, PortalSettings settings, String controlKey, String language, String[] additionalParameters)
 at DotNetNuke.Common.Globals.NavigateURL(Int32 tabID, Boolean isSuperTab, PortalSettings settings, String controlKey, String[] additionalParameters)
 at DotNetNuke.Common.Globals.NavigateURL(Int32 tabID, PortalSettings settings, String controlKey, String[] additionalParameters)
 at ToSic.SexyContent.EAV.Implementation.ValueConverter.SexyContentValueConverter.TryToResolveDnnCodeToLink(String value) in C:\Projects\2SexyContent\Web\DesktopModules\ToSIC_SexyContent\SexyContent\EAV\Implementation\ValueConverter\SexyContentValueConverter.cs:line 97
 at ToSic.Eav.Data.Entity.GetBestValue(String attributeName, String[] dimensions, Boolean resolveHyperlinks) in C:\Projects\eav-server\ToSic.Eav.Core\Entity.cs:line 222
 at ToSic.SexyContent.DynamicEntity.GetEntityValue(String attributeName, Boolean& propertyNotFound) in C:\Projects\2SexyContent\Web\DesktopModules\ToSIC_SexyContent\SexyContent\DynamicEntity.cs:line 83
 at ToSic.SexyContent.DynamicEntity.TryGetMember(String memberName, Object& result) in C:\Projects\2SexyContent\Web\DesktopModules\ToSIC_SexyContent\SexyContent\DynamicEntity.cs:line 69
 at CallSite.Target(Closure , CallSite , Object )
 at ASP._Page_Portals_2_2sxc_2sxcNews__List_News_cshtml.Execute() in c:\Projects\Home Instead 2016 Web\Web\Portals\2\2sxc\2sxcNews\_List-News.cshtml:line 70
 at System.Web.WebPages.WebPageBase.ExecutePageHierarchy()
 at System.Web.WebPages.WebPageBase.ExecutePageHierarchy(WebPageContext pageContext, TextWriter writer, WebPageRenderingBase startPage)
 at ToSic.SexyContent.Engines.RazorEngine.RenderTemplate() in C:\Projects\2SexyContent\Web\DesktopModules\ToSIC_SexyContent\2Sexy Content Razor\RazorEngine.cs:line 75
 at ToSic.SexyContent.Engines.EngineBase.Render() in C:\Projects\2SexyContent\Web\DesktopModules\ToSIC_SexyContent\SexyContent\Engines\EngineBase.cs:line 89
 at ToSic.SexyContent.SexyViewContentOrApp.ProcessView(PlaceHolder phOutput, Panel pnlError) in C:\Projects\2SexyContent\Web\DesktopModules\ToSIC_SexyContent\SexyContent\SexyViewContentOrApp.cs:line 136
```

**Comment by iJungleboy** [comment]
Just fyi: someone else had a similar problem - I think it was #746 - and by fixing it I think this is also fixed.

**Comment by raphael-m** [comment]
This is already fixed with commit ef67e1130afe51e4540830a11340983352b3ad48
